SyncTests/RecordTests.swift

```diff
@@ -27,6 +27,34 @@ class RecordTests: XCTestCase {
         XCTAssertTrue(json["ttl"].isNull())
     }
 
+    func testParsedNulls() {
+        // Make this a thorough test: use a real-ish blob of JSON.
+        let fullRecord = "{\"id\":\"global\"," +
+            "\"payload\":" +
+            "\"{\\\"syncID\\\":\\\"zPSQTm7WBVWB\\\"," +
+            "\\\"declined\\\":[\\\"bookmarks\\\"]," +
+            "\\\"storageVersion\\\":5," +
+            "\\\"engines\\\":{" +
+            "\\\"clients\\\":{\\\"version\\\":1,\\\"syncID\\\": null}," +
+            "\\\"tabs\\\":{\\\"version\\\":1,\\\"syncID\\\":\\\"W4H5lOMChkYA\\\"}}}\"," +
+            "\"username\":\"5817483\"," +
+        "\"modified\":1.32046073744E9}"
```

Comment:
Are there any other parts that could be null that we need to check? What if `clients` is null rather than a dictionary (again thinking of the lack of a null check in `isDictionary()`?